components/Dialogs/src/MDCDialogTransitionController.m

```diff
@@ -60,30 +71,60 @@ - (void)animateTransition:(id<UIViewControllerContextTransitioning>)transitionCo
   CGFloat endingAlpha = presenting ? 1 : 0;
 
   animatingView.frame = [transitionContext finalFrameForViewController:animatingViewController];
-  animatingView.alpha = startingAlpha;
 
-  NSTimeInterval transitionDuration = [self transitionDuration:transitionContext];
   UIViewAnimationOptions options =
       UIViewAnimationOptionAllowUserInteraction | UIViewAnimationOptionBeginFromCurrentState;
 
-  [UIView animateWithDuration:transitionDuration
-      delay:0.0
-      options:options
-      animations:^{
-        animatingView.alpha = endingAlpha;
-      }
-      completion:^(__unused BOOL finished) {
-        // If we're dismissing, remove the presented view from the hierarchy
-        if (!presenting) {
-          [fromView removeFromSuperview];
-        }
-
-        // From ADC : UIViewControllerContextTransitioning
-        // When you do create transition animations, always call the
-        // completeTransition: from an appropriate completion block to let UIKit know
-        // when all of your animations have finished.
-        [transitionContext completeTransition:YES];
-      }];
+  // Opacity animation
+  animatingView.alpha = startingAlpha;
+  [UIView animateWithDuration:_animationDuration / 2
+                        delay:0
+                      options:options
+                   animations:^{
+                     animatingView.alpha = endingAlpha;
+                   }
+                   completion:nil];
+
+  // Scale animation
+  MDCDialogPresentationController *presentationController = nil;
+  if ([animatingViewController.presentationController
+          isKindOfClass:[MDCDialogPresentationController class]]) {
+    presentationController =
+        (MDCDialogPresentationController *)animatingViewController.presentationController;
+  }
+
+  CGAffineTransform startingTransform =
+      presenting ? CGAffineTransformMakeScale(_dialogScaleFactor, _dialogScaleFactor)
```

Comment:
Nit: Please use `self.dialogScaleFactor` here rather than accessing the ivar directly.  [Accessing properties via accessor methods even within an object is an Apple best practice](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/EncapsulatingData/EncapsulatingData.html), unless it creates specific problems (e.g., from `init` or the accessor itself).

Reply:
Done.